java/vector/src/main/java/org/apache/arrow/vector/ipc/JsonFileReader.java

```diff
@@ -269,7 +269,7 @@ protected ArrowBuf read(BufferAllocator allocator, int count) throws IOException
 
     BufferReader INT8 = new BufferReader() {
       @Override
-      protected ArrowBuf read(BufferAllocator allocator, int count) throws IOException {
+      protected ArrowBuf read(BufferAllocator allocator, int count, int byteWidth) throws IOException {
```

Comment:
Hmm... I am not sure about this. Having `byteWidth` in every reader seems weird because most don't need it. Can you put make a member variable of fixed size binary reader?